pkg/templates/latesttag/template.go

```diff
@@ -40,6 +40,20 @@ func init() {
 				blockedRegexes = append(blockedRegexes, rg)
 			}
 
+			allowedRegexes := make([]*regexp.Regexp, 0, len(p.AllowList))
+			for _, res := range p.AllowList {
+				rg, err := regexp.Compile(res)
+				if err != nil {
+					return nil, errors.Wrapf(err, "invalid regex %s", res)
+				}
+				allowedRegexes = append(allowedRegexes, rg)
+			}
+
+			if len(blockedRegexes) > 0 && len(allowedRegexes) > 0 {
```

Comment:
Why this constraint? We can allow both to be set, and allow `blocked` to take precedence if something is both in allowed and blocked? The following semantics seems reasonable to me:
1. If allowed is empty, then everything is allowed automatically unless it is blocked.
2. If allowed is not empty, then only things that match an allowedRegex are allowed. Even for those things, if it matches a regex in blocked, then it is not allowed.
3. If blocked is empty, then the decision is entirely determined based on the allowedRegexes.
4. If blocked is not empty, then anything that matches a blocked regex is blocked, even if it's explicitly allowed.

Reply:
I've improved the check to implement the above suggested semantics. Thank you for pointing!

Reply:
In the original issue we discussed this and decided to have only 1 list to be set at at time to avoid unintentional issues with config.  There should be nothing wrong with users defining multiple checks for allow and deny vs putting in 1 check.  This alleviates any potential issues with config and allows the check to be single focused and should make it less error prone.

Reply:
Hmm I prefer this because it makes the configuration more readable. Often a user will have an intent like "allow only images from quay.io/my-org-name but do _not_ allow quay.io/my-org-name/image-name-used-for-dev". It would be nice to capture that in one check. This approach of having both an allowlist and denylist, and having one take precedence, is pretty common too -- for example, Kubernetes does this with add and drop capabilities.